File: SoftCapsuleCare/src/pe/edu/pucp/citamedica/pago/model/Pago.java
package pe.edu.pucp.citamedica.pago.model;

import java.util.Date;

public class Pago {
    private static int contador = 0;
    private int idPago;
    private int DNI;
    private double descuentoPorSeguro;
    private double montoParcial;
    private double montoTotal;
    private Date fechaPago;
    private String concepto;
    private boolean estado;

    public Pago() {
    }

    public Pago(int idPago, int DNI, double descuentoPorSeguro,double montoParcial, double montoTotal,Date fechaPago,
    String concepto,boolean estado) {
        this.idPago = ++contador;
        this.DNI = DNI;
        this.descuentoPorSeguro = descuentoPorSeguro;
        this.montoParcial = montoParcial;
        this.montoTotal = montoTotal;
        this.fechaPago = fechaPago;
        this.concepto = concepto;
        this.estado = false;
    }

    public int getIdPago() {
        return idPago;
    }

    public int getDNI() {
        return DNI;
    }

    public void setDNI(int DNI) {
        this.DNI = DNI;
    }

    public double getDescuentoPorSeguro() {
        return descuentoPorSeguro;
    }

    public void setDescuentoPorSeguro(double descuentoPorSeguro) {
        this.descuentoPorSeguro = descuentoPorSeguro;
    }

    public double getMontoTotal() {
        return montoTotal;
    }

    public void setMontoTotal(double montoTotal) {
        this.montoTotal = montoTotal;
    }

    public double getMontoParcial() {
        return montoParcial;
    }

    public void setMontoParcial(double montoParcial) {
        this.montoParcial = montoParcial;
    }

    public Date getFechaPago(){
        return fechaPago;
    }

    public void setFechaPago(Date fechaPago){
        this.fechaPago = fechaPago;
    }

    public String getConcepto(){
        return concepto;
    }

    public void setConcepto(String concepto){
        this.concepto = concepto;
    }

    public boolean getEstado(){
        return estado;
    }

    public void setEstado(boolean estado){
        this.estado = estado;
    }
    
    public void generarComprobantePago(){
        
    }
}

File: ClinicaManagement/ClinicaManagement/ClinicaModel/Pago.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClinicaModel
{

    public class Pago
    {
        private static int contador = 0;
        private int idPago;
        private double descuentoSeguro;
        private double montoParcial;
        private double montoTotal;
        private DateTime fechaPago;
        private string concepto;
        private bool estado;
        private int idPaciente;

        public int IdPago { get; set; }
        public double DescuentoSeguro { get; set; }
        public double MontoParcial { get; set; }
        public double MontoTotal { get; set; }
        public DateTime FechaPago { get; set; }
        public string Concepto { get; set; }
        public bool Estado { get; set; }
        public int IdPaciente { get; set; }

        public Pago()
        {
            IdPago = ++contador;
        }

        public Pago(double descuentoSeguro, double montoParcial, double montoTotal, DateTime fechaPago, string concepto, bool estado, int idPaciente)
        {
            IdPago = ++contador;
            DescuentoSeguro = descuentoSeguro;
            MontoParcial = montoParcial;
            MontoTotal = montoTotal;
            FechaPago = fechaPago;
            Concepto = concepto;
            Estado = estado;
            IdPaciente = idPaciente;
        }
    }
}
